lib/liquid/variable.rb

```diff
@@ -106,16 +106,20 @@ def evaluate_filter_expressions(context, filter_args, filter_kwargs)
       parsed_args
     end
 
-    def taint_check(obj)
-      if obj.tainted?
+    def taint_check(context, obj)
+      if obj.tainted? && Template.taint_mode != :lax
```

Comment:
small style nitpick... I would prefer this method to `return unless obj.tainted?` and `return if Template.taint_mode == :lax` rather than this big whole-method-if-block.